tests/model_formsets/tests.py

```diff
@@ -1655,6 +1655,69 @@ def test_prevent_change_outer_model_and_create_invalid_data(self):
         # created.
         self.assertQuerysetEqual(Author.objects.all(), ['<Author: Charles>', '<Author: Walt>'])
 
+    def test_validation_without_id(self):
+        AuthorFormSet = modelformset_factory(Author, fields='__all__')
+        data = {
+            'form-TOTAL_FORMS': '1',
+            'form-INITIAL_FORMS': '1',
+            'form-MAX_NUM_FORMS': '',
+            'form-0-name': 'Charles',
+        }
+        formset = AuthorFormSet(data=data)
+        self.assertFalse(formset.is_valid())
```

Comment:
I think the `is_valid()` assertions can be removed. The `formset.errors` assertions will fail if the formset is valid.